invenio_drafts_resources/services/records/components/media_files.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020-2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records service component base classes."""
+from invenio_i18n import gettext as _
+from invenio_records_resources.services.records.components import (
+    MediaFilesOptionsComponent,
+)
+from marshmallow import ValidationError
+
+from .files import DraftFilesComponent
+
+
+class DraftMediaFilesComponent(DraftFilesComponent):
```

Comment:
Maybe is better to make the `MediaFilesOptionsComponent` into mixins that just provide the attribute values. The following suggestion would mean that both inherit `ServiceComponent` (redundant) and otherwise we are redefining attributes that are already defined in records-resources (also redundant)

```suggestion
class DraftMediaFilesComponent(DraftFilesComponent, MediaFilesOptionsComponent):
```